tests/utilities/test_argparse.py

```diff
@@ -1,17 +1,51 @@
 import io
-from argparse import ArgumentParser
+from argparse import ArgumentParser, Namespace
 from typing import List
+from unittest.mock import MagicMock
 
 import pytest
 
 from pytorch_lightning import Trainer
 from pytorch_lightning.utilities.argparse import (
     add_argparse_args,
+    from_argparse_args,
     get_abbrev_qualified_cls_name,
+    parse_argparser,
     parse_args_from_docstring,
+    _gpus_arg_default,
+    _int_or_float_type
 )
 
 
+class ArgparseExample:
+    def __init__(self, a: int = 0, b: str = '', c: bool = False):
+        self.a = a
+        self.b = b
+        self.c = c
+
+
+def test_from_argparse_args():
+    args = Namespace(a=1, b='test', c=True, d='not valid')
+    my_instance = from_argparse_args(ArgparseExample, args)
+    assert my_instance.a == 1
+    assert my_instance.b == 'test'
+    assert my_instance.c
+
+    parser = ArgumentParser()
+    mock_trainer = MagicMock()
+    _ = from_argparse_args(mock_trainer, parser)
+    mock_trainer.parse_argparser.assert_called_once_with(parser)
+
+
+def test_parse_argparser():
+    args = Namespace(a=1, b='test', c=None, d='not valid')
+    new_args = parse_argparser(ArgparseExample, args)
+    assert new_args.a == 1
+    assert new_args.b == 'test'
+    assert new_args.c
```

Comment:
shouldn't this be?

```suggestion
    assert new_args.c is None
```

Reply:
`new_args.c` should have the value `True`, so the equivalent would be `assert new_args.c is True` if preferred?

Reply:
I didn't get why it should have the value `True` but it's explained here:

https://github.com/PyTorchLightning/pytorch-lightning/blob/42a7b7058573bc659eb1fd6a64035ae80e270211/pytorch_lightning/utilities/argparse.py#L64-L72

Reply:
Ah, that makes sense. I had forgotten about that :)